posix/unix.c

```diff
@@ -373,7 +372,14 @@ int unix_accept4(unsigned socket, struct sockaddr *address, socklen_t *address_l
 			break;
 		}
 
-		if ((new = unixsock_alloc(&newid, s->type, nonblock)) == NULL) {
+		if (s->nonblock && (s->connect == NULL)) {
+			err = -EWOULDBLOCK;
+			break;
+		}
+
+		/* alloc new socket with 2 refs: first ref for presence in the tree, second
+		 * ref returned to the caller */
+		if ((new = unixsock_alloc(&newid, s->type, nonblock, 2)) == NULL) {
```

Comment:
MISRA guideline. Please use this style here and elsewhere:
x = f(...)
if (x == NULL) {
  ...
}